com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java

```diff
@@ -480,97 +483,127 @@ static public void addToPaths(String dir) {
-        options.addOption(shortOpt, longOpt, hasArg, description);
+    public static final class CommandLineOption {
+        public final String shortOpt;
+        public final String longOpt;
+        public final ArgCount argCount;
+        public final ArgType argType;
+        public final String description;
+
+        public CommandLineOption(String shortOpt, String longOpt, ArgCount argCount, ArgType argType, String description) {
+            this.shortOpt = shortOpt;
+            this.longOpt = longOpt;
+            this.argCount = argCount;
+            this.argType = argType;
+            this.description = description;
+        }
+
+        public enum ArgCount {ZERO, ONE, MANY}
+
+        public enum ArgType {UNSPECIFIED, ABS_OR_CWD_REL_PATH}
+    }
+
+    private static CommandLineOption opt(String shortOpt, String longOpt, CommandLineOption.ArgCount argCount, CommandLineOption.ArgType argType, String description,
+                                         boolean usedByResourceCacheKey) {
         if (usedByResourceCacheKey) {
             ResourceCacheKey.includeOption(longOpt);
         }
+        if (argCount == MANY) {
+            description = description + ". More than one occurrence is allowed";
+        }
+        return new CommandLineOption(shortOpt, longOpt, argCount, argType, description);
+    };
+
+    private static CommandLineOption opt(String shortOpt, String longOpt, CommandLineOption.ArgCount argCount, String description, boolean usedByResourceCacheKey) {
+        return opt(shortOpt, longOpt, argCount, CommandLineOption.ArgType.UNSPECIFIED, description, usedByResourceCacheKey);
     }
 
-    public static Options getCommandLineOptions() {
-        Options options = new Options();
-        addOption(options, "r", "root", true, "Build root directory. Default is current directory", true);
-        addOption(options, "o", "output", true, "Output directory. Default is \"build/default\"", false);
-        addOption(options, "i", "input", true, "DEPRECATED! Use --root instead", true);
-        addOption(options, "v", "verbose", false, "Verbose output", false);
-        addOption(options, "h", "help", false, "This help message", false);
-        addOption(options, "a", "archive", false, "Build archive", false);
-        addOption(options, "ea", "exclude-archive", false, "Exclude resource archives from application bundle. Use this to create an empty Defold application for use as a build target", false);
-        addOption(options, "e", "email", true, "User email", false);
-        addOption(options, "u", "auth", true, "User auth token", false);
+    public static List<CommandLineOption> getCommandLineOptions() {
+        return List.of(
+                opt("r", "root", ONE, ABS_OR_CWD_REL_PATH, "Build root directory. Default is current directory", true),
+                opt("o", "output", ONE, "Output directory. Default is \"build/default\"", false),
```

Comment:
Great question! The answer is no for both options:
- `output` should not use `ABS_OR_CWD_REL_PATH`, because it is required to be a relative path, and it's resolved against project root by bob itself, so the editor shouldn't attempt to un-relativize it at all. See: https://github.com/defold/defold/blob/6fd84fb0a19e0416fd8e81b93b3e640ee2e02f46/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/fs/AbstractFileSystem.java#L53
- supplying `input` option leads to bob aborting the build altogether, so it does not matter what's the value of the option: https://github.com/defold/defold/blob/e184ca0b3fbc7c877e5872844d7e8470646a557f/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java#L781-L784